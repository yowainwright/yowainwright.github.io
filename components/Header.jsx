import React, { useContext } from 'react'
import Link from 'next/link'
import { Moon } from './svg/Moon'
import { Sun } from './svg/Sun'
import { DispatchStore, GlobalState } from '../pages/_app'

export const navItems = [
  {
    alias: 'resume',
    name: 'Resume',
    path: '/resume/',
  },
  {
    alias: 'archive',
    name: 'Archive',
    path: '/archive/',
  },
  {
    alias: 'about',
    name: 'About',
    path: '/about/',
  },
]

export const Icon = ({ isDarkMode }) => (isDarkMode ? <Sun /> : <Moon />)

export function DarkmodeToggle() {
  const state = useContext(GlobalState)
  const dispatch = useContext(DispatchStore)
  function handleToggle() {
    dispatch({ type: 'SET_IS_DARKMODE', payload: !state?.isDarkMode })
  }

  return (
    <button className='site-nav__toggle' onClick={handleToggle} title='Toggle Darkmode'>
      <Icon isDarkMode={state?.isDarkMode} />
    </button>
  )
}

export const NavItem = ({ alias, componentName, name, path }) => (
  <li className={`${componentName}__item ${componentName}__item--${alias}`}>
    <Link className={`${componentName}__link ${componentName}__link--${alias}`} href={path}>
      {name}
    </Link>
  </li>
)

export function NavList({ componentName, navItems }) {
  return (
    <ul className={`${componentName}__items`}>
      {navItems.map(({ alias, name, path }, i) => (
        <NavItem key={i} alias={alias} componentName={componentName} name={name} path={path} />
      ))}
    </ul>
  )
}

export function Header() {
  return (
    <nav id='site-nav' className='site-nav' role='navigation' itemType='http://schema.org/SiteNavigationElement'>
      <section className='site-nav__container'>
        <Link href='/' className='logo'>
          <h3 className='logo__title'>j</h3>
        </Link>
        <div className='site-nav__links-wrapper'>
          <NavList componentName='site-nav' navItems={navItems} />
          <DarkmodeToggle />
        </div>
      </section>
    </nav>
  )
}

export default Header
