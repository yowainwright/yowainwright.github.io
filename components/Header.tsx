import React, { useContext } from "react";
import Link from "next/link";
import { Moon } from "./svg/Moon";
import { Sun } from "./svg/Sun";
import { DispatchStore, GlobalState } from "../pages/_app";
import { NavItemProps, IconProps, NavListProps } from "../types";
import { NAV_ITEMS as navItems } from "../constants";

export const Icon = ({ isDarkMode }: IconProps) =>
  isDarkMode ? <Sun /> : <Moon />;

export function DarkmodeToggle() {
  const state = useContext(GlobalState);
  const dispatch = useContext(DispatchStore);
  const handleToggle = () =>
    dispatch({ type: "SET_IS_DARKMODE", payload: !state?.isDarkMode });

  return (
    <button
      className="site-nav__toggle"
      onClick={handleToggle}
      title="Toggle Darkmode"
    >
      <Icon isDarkMode={state?.isDarkMode} />
    </button>
  );
}

export const NavItem = ({ alias, componentName, name, path }: NavItemProps) => (
  <li className={`${componentName}__item ${componentName}__item--${alias}`}>
    <Link
      className={`${componentName}__link ${componentName}__link--${alias}`}
      href={path}
    >
      {name}
    </Link>
  </li>
);

export function NavList({ componentName, navItems }: NavListProps) {
  return (
    <ul className={`${componentName}__items`}>
      {navItems.map(({ alias, name, path }, i) => (
        <NavItem
          key={i}
          alias={alias}
          componentName={componentName}
          name={name}
          path={path}
        />
      ))}
    </ul>
  );
}

export function Header() {
  return (
    <nav
      id="site-nav"
      className="site-nav"
      role="navigation"
      itemType="http://schema.org/SiteNavigationElement"
    >
      <section className="site-nav__container">
        <Link href="/" className="logo">
          <h3 className="logo__title">j</h3>
        </Link>
        <div className="site-nav__links-wrapper">
          <NavList componentName="site-nav" navItems={navItems} />
          <DarkmodeToggle />
        </div>
      </section>
    </nav>
  );
}

export default Header;
