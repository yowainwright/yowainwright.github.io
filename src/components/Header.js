import React from "react"

class Header extends React.Component {
  render() {
    return (

    )
  }
}

export default Header
