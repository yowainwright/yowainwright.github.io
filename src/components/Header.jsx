import React, { Component } from 'react'

/*
  Header ğŸ‘¤
  ----
  The Header Component is the navigation and top part of the site across all pages/routes
*/
class Header extends Component {
  constructor (props) {
    super(props)
    this.name = 'site-nav'
  }

  /*
    TODO
    ----
    Abstract navItems out to a config
  */
  generateNav (name) {
    const nav = []

    // [array] of {object} nav items
    const navItems = [
      {
        alias: 'home',
        name: 'Jeffry.in',
        path: '/',
      },
      {
        alias: 'github',
        name: 'Github',
        path: 'https://github.com/yowainwright',
      },
      {
        alias: 'twitter',
        name: 'Twitter',
        path: 'https://twitter.com/yowainwright',
      },
      {
        alias: 'archive',
        name: 'Archive',
        path: '/archive/',
      },
      {
        alias: 'about',
        name: 'About',
        path: '/about/',
      },
    ]

    // build the nav list
    navItems.map((item, i) => {
      nav.push(
        <li key={i} className={`${name}__item ${name}__item--${item.alias}`}>
          <a className={`${name}__link ${name}__link--${item.alias}`} href={item.path}>{item.name}</a>
        </li>
      )
    })

    return nav
  }

  render () {
    return (
      <nav id={`${this.name}`} className={`${this.name}`} role='navigation' itemType='http://schema.org/SiteNavigationElement'>
        <ol className={`${this.name}__items`}>
          {this.generateNav(this.name)}
        </ol>
      </nav>
    )
  }
}

export default Header
