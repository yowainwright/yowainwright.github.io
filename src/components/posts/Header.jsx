import React, { Component } from 'react'

import Link from 'gatsby-link'

class Header extends Component {
  constructor(props) {
    super(props)
    this.path = props.path
    this.title = props.title
  }

  render() {
    return (
      <header className="post__header">
        <h2 className="post__title">
          <Link to={this.path}>{this.title}</Link>
        </h2>
        <time>{this.date}</time>
      </header>
    )
  }
}


export default Header
