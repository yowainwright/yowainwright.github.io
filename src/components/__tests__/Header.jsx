import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../Header'

describe('Header component', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<Header />).toJSON()

    expect(rendered).toMatchSnapshot()
  })
})
