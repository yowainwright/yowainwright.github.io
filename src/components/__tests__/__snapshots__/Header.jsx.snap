// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<nav
  className="site-nav"
  id="site-nav"
  itemType="http://schema.org/SiteNavigationElement"
  role="navigation"
>
  <div
    className="site-nav__container"
  >
    <Link
      className="site-nav__main-item-link"
      to="/"
    >
      <div
        className="site-nav__main-item"
      >
        <h3
          className="site-nav__main-item-symbol"
        >
          j
        </h3>
        <h2
          className="site-nav__main-item-title"
        >
          Jeffry.in
        </h2>
      </div>
    </Link>
    <ol
      className="site-nav__items"
    >
      <li
        className="site-nav__item site-nav__item--about"
      >
        <Link
          className="site-nav__link site-nav__link--about"
          to="/about/"
        >
          About
        </Link>
      </li>
      <li
        className="site-nav__item site-nav__item--archive"
      >
        <Link
          className="site-nav__link site-nav__link--archive"
          to="/archive/"
        >
          Archive
        </Link>
      </li>
    </ol>
  </div>
</nav>
`;
